src/components/Select/Select.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { FC, useState, useEffect, Ref } from 'react';
+
+import { mergeClasses } from '../../shared/utilities';
+import { Dropdown } from '../Dropdown';
+import { Menu } from '../Menu';
+import { TextInput, TextInputProps, TextInputWidth } from '../Inputs';
+import { Pill, PillProps, PillSize } from '../Pills';
+import { IconName } from '../Icon';
+import { SelectOption, SelectProps } from './Select.types';
+
+import styles from './select.module.scss';
+import { Spinner, SpinnerSize } from '../Spinner';
+
+export const Select: FC<SelectProps> = React.forwardRef(
+    (
+        {
+            options: _options = [],
+            inputWidth = TextInputWidth.fill,
+            defaultValue,
+            filterable = false,
+            multiple = false,
+            textInputProps = {},
+            dropdownProps = {},
+            pillProps = {},
+            menuProps = {},
+            loadOptions,
+            isLoading,
+            spinner = <Spinner size={SpinnerSize.Small} />,
+            classNames,
+            style,
+            'data-test-id': dataTestId,
+        },
+        ref: Ref<HTMLDivElement>
+    ) => {
+        const [dropdownVisible, setDropdownVisibility] =
+            useState<boolean>(false);
+        const [options, setOptions] = useState<SelectOption[]>(
```

Comment:
How will options get updated if someone updates the options props ?

Reply:
this is supported in the dynamic options user story. added a story for the same